libraries/resource_providers/api500/c7000/volume_provider.rb

```diff
@@ -20,33 +20,29 @@ def load_resource_with_associated_resources
           else
             validate_required_properties(:storage_system, :storage_pool)
             storage_system_uri = load_storage_system
-            @item['storagePoolUri'] = load_resource(:StoragePool, { name: @new_resource.storage_pool, storageSystemUri: storage_system_uri }, 'uri')
-            @item['snapshotPoolUri'] = load_resource(:StoragePool, { name: @new_resource.snapshot_pool, storageSystemUri: storage_system_uri }, 'uri') if @new_resource.snapshot_pool
+            @item.set_storage_pool(resource_named(:StoragePool).new(@item.client, name: @new_resource.storage_pool, storageSystemUri: storage_system_uri))
+            @item.set_snapshot_pool(resource_named(:StoragePool).new(@item.client, name: @new_resource.snapshot_pool, storageSystemUri: storage_system_uri)) if @new_resource.snapshot_pool
           end
-          set_properties unless @item.exists?
+          @item['properties']['name'] = @new_resource.name
+          @item.exists? ? @item.data.delete('properties') : set_properties
         end
 
         def load_storage_system
-          data = {
-            hostname: @new_resource.storage_system,
-            name: @new_resource.storage_system
-          }
-          load_resource(:StorageSystem, data, 'uri')
+          load_resource(:StorageSystem, { hostname: @new_resource.storage_system, name: @new_resource.storage_system }, 'uri')
```

Comment:
I did not drop this method as requested on provider 200 because I call it in more than one location, so I thought it would be best to keep it for a single point of service.